dracclient/tests/test_raid.py

```diff
@@ -112,6 +113,204 @@ def setUp(self):
+        expected_string_attr = raid.RAIDStringAttribute(
+            name='Name',
+            instance_id='Disk.Virtual.1:RAID.Integrated.1-1:Name',  # noqa
+            current_value='Virtual Disk 1',
+            pending_value=None,
+            read_only=True,
+            fqdd='Disk.Virtual.1:RAID.Integrated.1-1',
+            min_length=0,
+            max_length=129)
+        expected_integer_attr = raid.RAIDIntegerAttribute(
+            name='RAIDmaxSupportedVD',
+            instance_id='RAID.Integrated.1-1:RAIDmaxSupportedVD',  # noqa
+            current_value=240,
+            pending_value=None,
+            read_only=True,
+            fqdd='RAID.Integrated.1-1',
+            lower_bound=0,
+            upper_bound=0)
+        # expected_string_attr
+        mock_requests.post('https://1.2.3.4:443/wsman', [
+            {'text': test_utils.RAIDEnumerations[
+                uris.DCIM_RAIDEnumeration]['ok']},
+            {'text': test_utils.RAIDEnumerations[
+                uris.DCIM_RAIDString]['ok']},
+            {'text': test_utils.RAIDEnumerations[
+                uris.DCIM_RAIDInteger]['ok']}
+        ])
+
+        raid_settings = self.drac_client.list_raid_controller_settings()
+        self.assertEqual(219, len(raid_settings))
+        # enumerable attribute
+        self.assertIn(
+            'RAID.Integrated.1-1:RAIDCurrentControllerMode',  # noqa
+            raid_settings)
+        self.assertEqual(expected_enum_attr.fqdd, raid_settings[
+                         'RAID.Integrated.1-1:RAIDCurrentControllerMode'].fqdd)  # noqa
+        # string attribute
+        self.assertIn(
+            'Disk.Virtual.1:RAID.Integrated.1-1:Name',  # noqa
+            raid_settings)
+        self.assertEqual(expected_string_attr.fqdd,
+                         raid_settings['Disk.Virtual.1:RAID.Integrated.1-1:Name'].fqdd)  # noqa
+        # integer attribute
+        self.assertIn(
+            'RAID.Integrated.1-1:RAIDmaxSupportedVD',  # noqa
+            raid_settings)
+        self.assertEqual(expected_integer_attr.fqdd, raid_settings[
+            'RAID.Integrated.1-1:RAIDmaxSupportedVD'].fqdd)  # noqa
+
+    @mock.patch.object(dracclient.client.WSManClient,
+                       'wait_until_idrac_is_ready', spec_set=True,
+                       autospec=True)
+    @mock.patch.object(dracclient.client.DRACClient,
+                       'get_raid_controller_mode',
+                       spec_set=True, autospec=True)
+    def test_get_raid_controller_mode(self, mock_requests,
+                                      mock_get_raid_controller_mode,
+                                      mock_wait_until_idrac_is_ready):
+
+        mock_requests.post('https://1.2.3.4:443/wsman', [
+            {'text': test_utils.RAIDEnumerations[
+                uris.DCIM_RAIDEnumeration]['ok']},
+            {'text': test_utils.RAIDEnumerations[
+                uris.DCIM_RAIDString]['ok']}])
```

Comment:
Don't we need a response for DCIM_RAIDInteger above?

Reply:
Yes, Chris, We don't need a response for DCIM_RAIDInteger. If we add it will not affect so I have added it.